File: backend/src/main/java/com/yatmk/app/persistence/repositories/local/BookRepository.java
package com.yatmk.app.persistence.repositories.local;

import org.springframework.stereotype.Repository;

import com.yatmk.app.persistence.models.local.Book;
import com.yatmk.app.persistence.repositories.config.AbstractRepository;

@Repository
public interface BookRepository extends AbstractRepository<Book, Long> {

}

File: back-end-nest/src/persistence/repositories/book.repository.ts
import { DataSource, Repository } from "typeorm";
import { BookEntity } from "../models/local/book.entity";
import { Injectable } from "@nestjs/common";
import { Page } from "../models/config/page.interface";

@Injectable()
export class BookRepository extends Repository<BookEntity> {

  constructor(private readonly dataSource: DataSource) {
    super(BookEntity, dataSource.createEntityManager());
  }

  async findPaginated(page: number, size: number): Promise<Page<BookEntity>> {
    const [data, total] = await this.findAndCount({
      skip: (page - 1) * size, // Offset calculation
      take: size, // Limit for each page
      order: { id: 'ASC' }, // Optional sorting
    });
    const totalPages = Math.ceil(total / size);
    return new Page(data, total, totalPages, page, size);
  }

}